### PR TITLE
[WIP] Explore string interpolation library

### PR DESCRIPTION
**Pull request by marler8997**
A prototype implementation for interpolated strings to explore their viability as a library.  Comments/suggestions welcome.

### CODE REVIEW

File: std/experimental/interpolate.d [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+writeln(mixin(interp("a is $(a)")));
+// same as: writeln(mixin(`tuple("a is ", a).expand`));
+
+// Output:
+// a is 42
+---
+*/
+string interp(string str, string file = __FILE__, size_t line = __LINE__) pure @safe
+{
+    return "tuple(" ~ interpToCommaExpression(str, file, line) ~ ").expand";
```
Comment (ntrel) [inline std/experimental/interpolate.d]: I think we could avoid the public import by (essentially) doing this:
```d
return q{() @property {import std.typecons : tuple; return tuple(...);}};
```

Comment (marler8997) [reply std/experimental/interpolate.d]: great idea...changed

**Comment by marler8997** [comment]
Cleaning up old PRs that aren't going anywhere

**Comment by Geod24** [comment]
@marler8997 : Could you possibly turn this into a dub package ?

**Comment by marler8997** [comment]
Yeah that's possible, the purpose of it was a proof of concept to see if interpolated strings should be implemented as a library or as part of the language.  If it works well as a library then we could make a dub package.  It has some problems though, for example if we want to support arbitrary expressions inside the string then we really need to include the DMD lexer...another reason why it would make sense to put it in the compiler instead of as a library.

**Pull request closed without merging**